contracts/SampleVerifier.sol

```diff
@@ -0,0 +1,253 @@
+        require(disputes[disputeId].state == States.Initialised, "game not initialised");
+        _;
+    }
+
+    /**
+      * @dev only run when a game in Initialised state
+      */
+    modifier onlyFoundDiff(bytes32 disputeId) {
+        require(disputes[disputeId].state == States.FoundDiff, "game not found diff");
+        _;
+    }
+
+    /**
+      * @dev only allow solver
+      */
+    modifier onlySolver(bytes32 disputeId) {
+        require(disputes[disputeId].solver == msg.sender, "sender not solver");
+        _;
+    }
+
+    /**
+      * @dev only allow challenger
+      */
+    modifier onlyChallenger(bytes32 disputeId) {
+        require(disputes[disputeId].challenger == msg.sender, "sender not challenger");
+        _;
+    }
+
+    /**
+      * @dev solver submits initial execution state proofs
+      */
+    function solverProofs(
+        bytes32 disputeId,
+        bytes32[] startProofs,
+        bytes32 startHash,
+        bytes32[] endProofs,
+        bytes32 endHash
+    ) public onlySolver(disputeId) onlyInitialised(disputeId) onlyPlaying(disputeId) {
+
+        Dispute storage dispute = disputes[disputeId];
+        if (MerkleProof.verify(startProofs, dispute.solverComputationHash.merkleRoot, startHash, 0) &&
+            MerkleProof.verify(endProofs, dispute.solverComputationHash.merkleRoot, endHash, dispute.solverComputationHash.stepCount - 1)) {
+            dispute.left = StateHash(startHash, 0);
+            dispute.right = StateHash(endHash, dispute.solverComputationHash.stepCount - 1);
+            dispute.state = States.SolverTurn;
+            dispute.timeout = getTimeout();
+        } else {
+            // solver lost immediately
+            dispute.state = States.Ended;
+            dispute.result = Results.ChallengerCorrect;
+            enforcer.result(dispute.executionId, false, dispute.challenger);
+        }
+    }
+
+    /**
+      * @dev run once found the different step of a dispute
+      */
+    function detailExecution(
+        bytes32 disputeId,
+        bytes code,
+        // bytes32[] codeProof,
+        bytes data,
+        uint256[4] params,
+        uint256[] stack,
+        bytes mem,
+        // uint256 memPos,
+        // bytes32[] memProof,
+        uint256[] accounts,
+        bytes accountsCode,
+        bytes32 logHash
+    ) public onlyFoundDiff(disputeId) onlyPlaying(disputeId) {
+        Dispute storage dispute = disputes[disputeId];
+
+        require(dispute.left.step == dispute.right.step - 1, "must find a specific step");
+
+        // calculate code root from code and codeProof, code pos is pcStart
+        // code is actually more complicated to workwith
+        // we may need 1 byte + 32 bytes of code to run 1 opcode
+        // calculate stack hash from stack and stackHash
+        // calculate mem hash root from mem and proof
+        // CompactMemory
+        // verify left state
+        require(stack.toHash(0) == dispute.left.hash, "state hash not match");
```

Comment:
you only check the stack, because https://github.com/leapdao/solEVM-enforcer/issues/4 still in progress?